extism.go

```diff
@@ -382,6 +399,18 @@ func NewPlugin(
 			data.Name = "main"
 		}
 
+		if data.Name == "main" && config.ObserveAdapter != nil {
+			// TODO: Since we can extract the names out of a api.Module,
+			// maybe we can have a new overload that accepts that instead of []byte?
+			trace, err = config.ObserveAdapter.NewTraceCtx(ctx, c.Wazero, data.Data, config.ObserveOptions)
```

Comment:
If we can get all the data we need, and there are no side effects to doing anything with the `api.Module`, then yea I think that makes sense! 

We could also consider just changing it from `[]byte` input to `api.Module`, since this SDK is always dependent on Wazero. I think we'd have another SDK or different API altogether if we ever were to have an alternative to Wazero emerge.